### ISSUE TITLE
Export to Google Sheets fails with "Doclist API has been shut down"

### ISSUE THREAD

**Issue by tfmorris** [OpenRefine/OpenRefine#996]
Creating a new spreadsheet now requires the use of the Drive API. 
https://developers.google.com/drive/v2/reference/

17:52:13.630 [             gdata_upload] The Doclist API has been shut down. Please use the new Drive API, https://developers.google.com/drive/. (405ms)
com.google.gdata.util.InvalidEntryException: The Doclist API has been shut down. Please use the new Drive API, https://developers.google.com/drive/.
<errors xmlns='http://schemas.google.com/g/2005'><error><domain>GData</domain><code>BadRequestException</code><internalReason>The Doclist API has been shut down. Please use the new Drive API, https://developers.google.com/drive/.</internalReason></error></errors>

	at com.google.gdata.client.http.HttpGDataRequest.handleErrorResponse(HttpGDataRequest.java:602)
	at com.google.gdata.client.http.GoogleGDataRequest.handleErrorResponse(GoogleGDataRequest.java:564)
	at com.google.gdata.client.http.HttpGDataRequest.checkResponse(HttpGDataRequest.java:560)
	at com.google.gdata.client.http.HttpGDataRequest.execute(HttpGDataRequest.java:538)
	at com.google.gdata.client.http.GoogleGDataRequest.execute(GoogleGDataRequest.java:536)
	at com.google.gdata.client.Service.insert(Service.java:1409)
	at com.google.gdata.client.GoogleService.insert(GoogleService.java:613)
	at com.google.gdata.client.media.MediaService.insert(MediaService.java:381)
	at com.google.refine.extension.gdata.UploadCommand.uploadSpreadsheet(UploadCommand.java:155)

**Comment by jackyq2015** [comment]
I think upgrading the gdata library can solve this problem:
https://developers.google.com/gdata/javadoc/

Another option is to introduce the apache camle to handle different import/export requirement. It also has the google drive component:
http://camel.apache.org/googledrive.html

Any comment?

**Comment by jackyq2015** [comment]
@tfmorris can you assign this ticket to me? Just make sure community members can focus on different ticket to prevent duplicate work.

Also I think this ticket depends on #970. You can assign that to me so you can put your energy on more important thing.

**Comment by tfmorris** [comment]
@jackyq2015 Thanks for the offer, but I've already got most of the work for #979 done (I'm assuming that's what you meant instead of 970).  It's a little bit complex because of the shared dependencies for Fusion Tables, Google Sheets, and Freebase, so I still need to do a little more testing.

I think there is more work required than just upgrading the libraries, but it's a necessary first step.

Note that Google Sheets import still works.  It's just the export which has stopped working.

p.s. As an aside, it's apparently not possible to assign an issue to a non-project member. I tried to assign another issue to you and wasn't able to do it, so we'll have to just rely on "claiming" issues using comments.

**Comment by jackyq2015** [comment]
@tfmorris Glad to hear that you got things fixed. Please Let me know if you find some 2.6 tickets that I can help with. I will also try to "claiming"issues from the 2.6 milestone.

Regarding this issue, I did some exploration to integrate the OpenRefine with Apache Camel. I would like to post my findings to the mailing list to get some feed back.

**Comment by tfmorris** [comment]
This is fixed, but it required removing/disabling the Freebase extension.  There's a Catch-22 because more recent versions of the Freebase client library don't include MQLRead and older versions don't work with current versions of the core Google client API libraries (and you can't have multiple versions of the dependent libraries easily).  We might be able to resovle this with fancy classloader tricks, but for the remaining few months of Freebase availability, it's not clear it's worth the effort.
